commit-event-service/src/main/scala/ch/datascience/commiteventservice/events/categories/commitsync/eventgeneration/CommitEventSynchronizer.scala

```diff
@@ -0,0 +1,258 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ch.datascience.commiteventservice.events.categories.commitsync.eventgeneration
+
+import cats.data.StateT
+import cats.effect.{ContextShift, IO, Timer}
+import cats.syntax.all._
+import cats.{Applicative, MonadThrow}
+import ch.datascience.commiteventservice.events.categories.commitsync.eventgeneration.CommitEventSynchronizer.SynchronizationSummary.{SummaryState, add}
+import ch.datascience.commiteventservice.events.categories.commitsync.eventgeneration.CommitEventSynchronizer.UpdateResult._
+import ch.datascience.commiteventservice.events.categories.commitsync.eventgeneration.CommitEventSynchronizer.{SynchronizationSummary, UpdateResult}
+import ch.datascience.commiteventservice.events.categories.commitsync.eventgeneration.historytraversal.{CommitInfoFinder, CommitToEventLog, EventDetailsFinder}
+import ch.datascience.commiteventservice.events.categories.commitsync._
+import ch.datascience.config.GitLab
+import ch.datascience.control.Throttler
+import ch.datascience.events.consumers.Project
+import ch.datascience.graph.model.events.{BatchDate, CommitId}
+import ch.datascience.graph.tokenrepository.AccessTokenFinder
+import ch.datascience.graph.tokenrepository.AccessTokenFinder._
+import ch.datascience.http.client.AccessToken
+import ch.datascience.logging.ExecutionTimeRecorder
+import ch.datascience.logging.ExecutionTimeRecorder.ElapsedTime
+import org.typelevel.log4cats.Logger
+
+import scala.concurrent.ExecutionContext
+import scala.util.control.NonFatal
+private[commitsync] trait CommitEventSynchronizer[Interpretation[_]] {
+  def synchronizeEvents(event: CommitSyncEvent): Interpretation[Unit]
+}
+
+private[commitsync] class CommitEventSynchronizerImpl[Interpretation[_]: MonadThrow](
+    accessTokenFinder:     AccessTokenFinder[Interpretation],
+    latestCommitFinder:    LatestCommitFinder[Interpretation],
+    eventDetailsFinder:    EventDetailsFinder[Interpretation],
+    commitInfoFinder:      CommitInfoFinder[Interpretation],
+    commitToEventLog:      CommitToEventLog[Interpretation],
+    commitEventsRemover:   CommitEventsRemover[Interpretation],
+    executionTimeRecorder: ExecutionTimeRecorder[Interpretation],
+    logger:                Logger[Interpretation],
+    clock:                 java.time.Clock = java.time.Clock.systemUTC()
+) extends CommitEventSynchronizer[Interpretation] {
+
+  import executionTimeRecorder._
+  override def synchronizeEvents(event: CommitSyncEvent): Interpretation[Unit] = (for {
+    maybeAccessToken  <- accessTokenFinder.findAccessToken(event.project.id)
```

Comment:
Maybe we can import members of the `accessToken` and `latestCommitFinder` so we can use the defs here directly for improved readability?